js/commit-page.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Javascript for project commit pages to update the status and build the left sidebar
+ */
+
+$(document).ready(function(){
+  var url_parts = window.location.href.split('?')[0].split('/');
+  var parent_url = url_parts.slice(0,-2).join('/');
+  var myCommitId = url_parts.slice(-1);
+
+  var request;
+  if(window.XMLHttpRequest)
+    request = new XMLHttpRequest();
+  else
+    request = new ActiveXObject("Microsoft.XMLHTTP");
```

Comment:
Since we're using jQuery, it might be better to use $.ajax() instead of this request syntax. See http://api.jquery.com/jquery.ajax/